fvm/fvm.go

```diff
@@ -51,9 +51,12 @@ func (vm *VirtualMachine) Run(ctx Context, proc Procedure, v state.View, program
 		meter.NewMeter(
 			uint(proc.ComputationLimit(ctx)),
 			uint(proc.MemoryLimit(ctx))),
-		state.WithMaxKeySizeAllowed(ctx.MaxStateKeySize),
-		state.WithMaxValueSizeAllowed(ctx.MaxStateValueSize),
-		state.WithMaxInteractionSizeAllowed(ctx.MaxStateInteractionSize))
+		state.StateParameters{
+			MaxKeySizeAllowed:     ctx.MaxStateKeySize,
+			MaxValueSizeAllowed:   ctx.MaxStateValueSize,
+			MaxInteractionAllowed: ctx.MaxStateInteractionSize,
+		},
```

Comment:
maybe still use `state.DefaultParameters().WithXXX` here to future-proof against additions of new parameters?

Reply:
Or mabe have a NewStateParameters function so that when a new state parameter is added this code fails to compile?